packages/scandipwa/src/route/ProductPage/ProductPage.component.js

```diff
@@ -132,31 +164,101 @@ export class ProductPage extends PureComponent {
         );
     }
 
-    renderAdditionalSections() {
+    renderProductInformationTab() {
         const {
             dataSource,
             parameters,
+            areDetailsLoaded,
+            isInformationTabEmpty
+        } = this.props;
+
+        if (isInformationTabEmpty) {
+            return null;
+        }
+
+        return (
+            <ProductInformation
+              product={ { ...dataSource, parameters } }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductAttributesTab() {
+        const {
+            dataSource,
+            parameters,
+            areDetailsLoaded,
+            isAttributesTabEmpty
+        } = this.props;
+
+        if (isAttributesTabEmpty) {
+            return null;
+        }
+
+        return (
+            <ProductAttributes
+              product={ { ...dataSource, parameters } }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductReviewsTab() {
+        const {
+            dataSource,
+            areDetailsLoaded
+        } = this.props;
+
+        return (
+            <ProductReviews
+              product={ dataSource }
+              areDetailsLoaded={ areDetailsLoaded }
+            />
+        );
+    }
+
+    renderProductTabItems() {
+        const productTabItems = [];
+        Object.values(this.tabMap).map((item) => {
+            productTabItems.push(item.render());
+            return true;
+        });
+
+        return productTabItems.filter(Boolean).map((item) => <div key={ item.toString() }>{ item }</div>);
```

Comment:
What is this monstrosity?

1. You define a new array
2. You get values of tabMap
3. You call MAP function and assign this new map to nothing
3. You push renders to array
4. You filter array by boolean
5. You map items again to div which key is render to string

Are you crazy?

The right way to do it here is:
```
return Object.values(this.tabMap).reduce((acc, { shouldTabRender, render }) => {
    if (shouldTabRender) {
        acc.push(render());
    }
}, []);
```

Hard-code keys right into the rendering methods of each tab, no need to wrap them into unnecessary DIVs